modin/core/execution/ray/implementations/pandas_on_ray/partitioning/virtual_partition.py

```diff
@@ -0,0 +1,487 @@
+        )
+
+    def _wrap_partitions(self, partitions):
+        """
+        Wrap partitions passed as a list of ``ray.ObjectRef`` with ``PandasOnRayDataframePartition`` class.
+
+        Parameters
+        ----------
+        partitions : list
+            List of ``ray.ObjectRef``.
+
+        Returns
+        -------
+        list
+            List of ``PandasOnRayDataframePartition`` objects.
+        """
+        return [
+            self.partition_type(object_id, length, width, ip)
+            for (object_id, length, width, ip) in zip(*[iter(partitions)] * 4)
+        ]
+
+    def apply(
+        self,
+        func,
+        num_splits=None,
+        other_axis_partition=None,
+        maintain_partitioning=True,
+        **kwargs,
+    ):
+        """
+        Apply a function to this axis partition along full axis.
+
+        Parameters
+        ----------
+        func : callable
+            The function to apply.
+        num_splits : int, default: None
+            The number of times to split the result object.
+        other_axis_partition : PandasDataframeAxisPartition, default: None
+            Another `PandasDataframeAxisPartition` object to be applied
+            to func. This is for operations that are between two data sets.
+        maintain_partitioning : bool, default: True
+            Whether to keep the partitioning in the same
+            orientation as it was previously or not. This is important because we may be
+            operating on an individual AxisPartition and not touching the rest.
+            In this case, we have to return the partitioning to its previous
+            orientation (the lengths will remain the same). This is ignored between
+            two axis partitions.
+        **kwargs : dict
+            Additional keywords arguments to be passed in `func`.
+
+        Returns
+        -------
+        list
+            A list of `PandasOnRayDataframeVirtualPartition` objects.
+        """
+        if not self.full_axis:
+            num_splits = 1
+        if len(self.call_queue) > 0:
+            self.drain_call_queue()
+        result = super(PandasOnRayDataframeVirtualPartition, self).apply(
+            func, num_splits, other_axis_partition, maintain_partitioning, **kwargs
+        )
+        if self.full_axis:
+            return result
+        else:
+            # must unpack subset of the axis to ensure correct dimensions on
+            # partitions object
+            return result[0]
+
+    def force_materialization(self, get_ip=False):
+        """
+        Materialize partitions into a single partition.
+
+        Parameters
+        ----------
+        get_ip : bool, default: False
+            Whether to get node ip address to a single partition or not.
+
+        Returns
+        -------
+        PandasOnRayDataframeVirtualPartition
+            An axis partition containing only a single materialized partition.
+        """
+        materialized = super(
```

Comment:
I'm not sure why this is called? The only two classes (row axis and col axis partition) that inherit from this class don't override this method?

Reply:
Sorry, I don't understand your comment. This is here as an optimization from the parent, where when we materialize, we can just change the list of partitions to the materialized partition underneath. The children classes will both inherit this.